docs/developer/test_suite.md

```diff
@@ -0,0 +1,465 @@
+
+```Shell
+    cd Test
+    make check
+```
+
+or
+
+```Shell
+    cmake -t check
+```
+
+following a compile with either the GNU make or cmake build systems. This will produce a brief summary output along the lines of
+
+```Shell
+make check COMMS_ARCH=mpi
+Found mpirun, using PARALLEL_CMD=mpirun -np tc.nprocs
+rm -f */*/*.castep */*/*.dfpt_wvfn */*/*.fd_wvfn */*/*.wvfn.* */*/*.*.err
+../bin/testcode.py -q  --processors=4 --total-processors=36 --user-option castep run_cmd_template "../../../bin/run_castep_test.pl tc.program tc.args tc.input tc.output mpirun -np tc.nprocs" -e /home/user/CASTEP-23.1/bin/castep.mpi -c simple -c d3-simple -c d4-simple -c md-parallel
+....................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................... [487/487]
+```
+
+if everything passes or with `F` replacing one or more of the `.` symbols to indicate a failed test or tests.
+
+In case of a failed test, more detailed information on which tests failed and why, the commands 
+`make compare`, `make diff`, `cmake -t compare`, or `cmake -t diff`
+will print a summary or detailed analysis of the failed tests to stdout.
+
+### Partial and incremental testing
+
+During a code development process, running the entire test suite to
+check every change will be too time consuming and functionally
+unnecessary.  The tests are divided into categories, named `spe`
+,`bs` ,`geom` ,`md` ,`magres` ,`misc`
+,`pair-pot` ,`phonon` ,`otf` ,`pseudo`
+,`tddft` ,`ts` ,`xc` ,`nlxc` ,`libxc`
+,`solvation` ,`d3` ,`d4` ,`python`
+and `parallel` corresponding roughly to the subdirectories
+
+```
+Test/
+  |-Electronic
+  |-Excitations
+  |-Geometry
+  |-MD
+  |-Magres
+  |-Misc
+  |-Pair-Pot
+  |-Phonon
+  |-Pseudo
+  |-TDDFT
+  |-XC
+  |-Solvation
+  |-Dispersion
+  |-Pseudopotentials
+  |-Python
+  |-Utilities
+  |-LibXC
+  |-Parallel
+```
+
+and may be run individually using
+
+```
+    make check-<category>
+    make compare-<category>
+    make diff-<category>
+```
+
+or the targets of the same name in cmake.
+
+??? keyword "Running individual tests"
+    In fact the tests may be run at even finer granularity, down to the subdirectory. This is not implemented via the Makefile or cmake, and can only be accessed by invoking `testcode` directly.  The most straightforward means of doing this is to copy and paste the testcode invocation command printed by `make check-spe` including all of the run and argument setup and change the argument of the `-c` option to `<directory>/<subdirectory>`. See section [Invoking Testcode](#direct-invocation-of-testcode)
```

Comment:
> The most straightforward means of doing this is to copy and paste the testcode invocation command printed by `make check-spe`

Provide it here instead? While potentially liable to change I doubt the command will change sufficiently frequently to not warrant putting it here. 

At the very least make it clear the link to below contains the requisite lines.